model/canvas.cc

```diff
@@ -145,9 +145,25 @@ namespace minsky
       }
     if (fromPort.get())
       {
-          if (auto to=closestInPort(x,y))
+          if (auto to=closestInPort(x,y)) {
             model->addWire(static_cast<shared_ptr<Port>&>(fromPort),to);
-        fromPort.reset();
+            fromPort.reset();
+            auto toItem = to.get()->item();
+            if (auto ravel=static_cast<Ravel&>(toItem))
+            {
+              auto state = ravel.getState();
+              if(state.empty()) {
+                minsky().reset();
+              }
+            } else if(auto sheet=static_cast<Sheet&>(toItem)) {
+              if(sheet.empty()) {
```

Comment:
If we're doing this, maybe we could define empty() to be a virtual method of Item.  empty() is perhaps the wrong name - maybe something like inputDefined() might be better. There is something like that already for variables, so that could be generalised.  This will obviate any requirement for dynamic_cast.